R/check_values.R

```diff
@@ -69,8 +71,15 @@ vec_check_values <- function(
   # Check if values are the same length
   return_if_problem(vec_check_dimensions(object, expected))
 
+  c(object, expected) %<-% vctrs::vec_cast_common(object, expected)
+
   # Check if values are the same
-  if (!all(vctrs::vec_equal(object, expected, na_equal = TRUE))) {
+  if (
+    !isTRUE(all.equal(
+      object, expected,
+      tolerance = tolerance, check.attributes = FALSE, check.names = FALSE
+    ))
+  ) {
```

Comment:
```suggestion
  obj_is_equal <- all.equal(
    object,
    expected,
    tolerance = tolerance,
    check.attributes = FALSE,
    check.names = FALSE
  )
  
  if (!isTRUE(obj_is_equal)) {
```

or alternatively add a helper function in `utils.R`

```r
is_all_equal <- function(x, y, tolerance = tolerance, check.attributes = FALSE, check.names = FALSE) {
  isTRUE(all.equal(x, y, tolerance = tolerance, check.attributes= check.attributes, check.names = check.names))
}
```

```suggestion
  if (is_all_equal(object, expected, tolerance = tolerance)) {
```

Actually, I think that's the best approach since it can also be used to remove the need for zeallot:

```r
is_all_equal <- function(x, y, tolerance = tolerance, check.attributes = FALSE, check.names = FALSE) {
  obj <- vctrs::vec_cast_common(x, y)
  eq <- all.equal(obj$x, obj$y, tolerance = tolerance, check.attributes= check.attributes, check.names = check.names)
  isTRUE(eq)
}
```

Reply:
Refactored based on your suggestion in [5b8d573](https://github.com/rstudio/tblcheck/pull/111/commits/5b8d5733ccba1a31e886d8e84faa04c17524b68d)

Reply:
Don't forget to remove the zeallot dependency

Reply:
Oops! [ee7218c](https://github.com/rstudio/tblcheck/pull/111/commits/ee7218c756e5357095232bb150c49f8483c10724)